http-server/src/main/java/org/triplea/server/http/IpAddressExtractor.java

```diff
@@ -0,0 +1,30 @@
+package org.triplea.server.http;
+
+import javax.servlet.http.HttpServletRequest;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+/**
+ * Utility class to extract the remote host machine IP address from an http servlet request.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class IpAddressExtractor {
```

Comment:
This functionality is now used in multiple places and was extracted.

Reply:
Good ol' `X-Forwarded-For` header ^^
Just saying that this would in theory allow to "modify" the users origin address, we might want to check both